btcpy/structs/address.py

```diff
@@ -50,9 +50,10 @@ def get_codec():
         return Base58Codec
 
     def __init__(self, addr_type, hashed_data, mainnet=None):
-        if mainnet is None:
-            mainnet = is_mainnet()
-        network = 'mainnet' if mainnet else 'testnet'
+        network = 'mainnet' if mainnet is True else None
+        network = 'testnet' if mainnet is False else network
```

Comment:
Readability might be improved here by explicitly checking the three cases (`mainnet is True`, `mainnet is False`, `mainnet is None`)